quinn-proto/src/crypto.rs

```diff
@@ -209,16 +210,13 @@ pub trait AeadKey {
     ) -> Result<&'a mut [u8], CryptoError>;
 }
 
-/// Generic crypto errors
-#[derive(Debug)]
-pub struct CryptoError;
-
-/// Error indicating that the specified QUIC version is not supported
-#[derive(Debug)]
-pub struct UnsupportedVersion;
-
-impl From<UnsupportedVersion> for ConnectError {
-    fn from(_: UnsupportedVersion) -> Self {
-        Self::UnsupportedVersion
-    }
+/// Crypto layer errors
+#[derive(Clone, Copy, Debug, Eq, Error, PartialEq)]
+pub enum CryptoError {
+    /// Error from the cryptographic session layer
+    #[error("crypto error")]
+    Other,
+    /// Specified QUIC version is not supported
+    #[error("unsupported QUIC protocol version")]
+    UnsupportedVersion,
```

Comment:
Why combine the errors? I don't see any new error cases in `initial_keys`, and it's nice to express that guarantee.

Reply:
Yeah, I think there was a reason when I was experimenting with getting this working against 0.22, but it doesn't seem to be needed anymore.